### PR TITLE
Generate: logits processors are doctested and fix broken doctests

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

This PR:
- Adds `logits_processors.py` to the doctests
- Fixes the broken tests (while fixing the tests, also improved the examples with better practices and shorter examples)
- Updates sample-related examples to much shorter examples (the examples were too long and not very representative of the processor)

### CODE REVIEW

File: utils/not_doctested.txt [hunk 1]
```diff
@@ -371,7 +371,6 @@ src/transformers/generation/beam_constraints.py
 src/transformers/generation/beam_search.py
 src/transformers/generation/flax_logits_process.py
 src/transformers/generation/flax_utils.py
-src/transformers/generation/logits_process.py
```
Comment (gante) [inline utils/not_doctested.txt:374]: @ydshieh this is enough to make the file part of the doctests, correct? :)

Comment (ydshieh) [reply utils/not_doctested.txt:374]: Yes!

**Review by ydshieh — APPROVED** [review]
Thanks!

**Comment by ydshieh** [comment]
It's nice to see the red cross 🔥

**Comment by gante** [comment]
ooh boy 🤦

**Comment by ArthurZucker** [comment]
Super glad to see the full power of the `tests_pr_documentation_tests` on display 🚀

**Comment by gante** [comment]
@ArthurZucker ready for a review! I've updated the PR header with a summary of the changes :)

File: src/transformers/generation/logits_process.py [hunk 2]
```diff
@@ -154,31 +154,24 @@ class MinNewTokensLengthLogitsProcessor(LogitsProcessor):
...
+    >>> # If the maximum length (default = 20) is smaller than the minimum length constraint, the latter is ignored!
+    >>> outputs = model.generate(**inputs, min_new_tokens=30)
     >>> print(tokenizer.decode(outputs[0], skip_special_tokens=True))
     Hugging Face Company is a company that has been working on a new product for the past year.
 
-    >>> # If `eos_token_id` is set to ` company` it will take into account how many `min_new_tokens` have been generated
-    >>> # before stopping. Note that ` Company` (5834) and ` company` (1664) are not actually the same token, and even
-    >>> # if they were ` Company` would be ignored by `min_new_tokens` as it excludes the prompt.
-    >>> outputs = model.generate(input_ids=input_ids, min_new_tokens=1, eos_token_id=1664)
+    >>> # For testing purposes, let's set `eos_token_id` to "company", the first generated token. This will make
```
Comment (ArthurZucker) [inline src/transformers/generation/logits_process.py:164]:
```suggestion
    >>> # For testing purposes, let's set `eos_token` to `"company"`, the first generated token. This will make
```
Let's be more specific hehe

File: src/transformers/generation/logits_process.py [hunk 3]
```diff
@@ -154,31 +154,24 @@ class MinNewTokensLengthLogitsProcessor(LogitsProcessor):
...
+    >>> # generation end there.
+    >>> outputs = model.generate(**inputs, eos_token_id=1664)
     >>> print(tokenizer.decode(outputs[0], skip_special_tokens=True))
     Hugging Face Company is a company
 
-    >>> # Increasing `min_new_tokens` will bury the first occurrence of ` company` generating a different sequence.
-    >>> outputs = model.generate(input_ids=input_ids, min_new_tokens=2, eos_token_id=1664)
+    >>> # Increasing `min_new_tokens` will make generation ignore occurences "company" (eos token) before the
```
Comment (ArthurZucker) [inline src/transformers/generation/logits_process.py:170]:
```suggestion
    >>> # Increasing `min_new_tokens` will make generation ignore occurences `"company"` (eos token) before the
```

**Review by ArthurZucker — APPROVED** [review]
Thanks!
